crud/src/test/java/com/redhat/lightblue/crud/mongo/AbstractMongoCrudTest.java

```diff
@@ -79,8 +80,8 @@ public void setup() throws Exception {
 
     @After
     public void teardown() throws Exception {
-        mongo.getConnection().dropDatabase(DATABADSE_NAME);
-        coll = null;
+        if(coll!=null)
+            coll.remove(new BasicDBObject());
```

Comment:
Dropping the database is done asynchronously. So after dropDatabase
returns, the next test starts running, and then the actual drop takes
place, screwing it up.

Removing eveything is synchronous.

On Thu, Jul 30, 2015 at 11:38 AM, Dennis Crissman <notifications@github.com>
wrote:

> In
> crud/src/test/java/com/redhat/lightblue/crud/mongo/AbstractMongoCrudTest.java
> <https://github.com/lightblue-platform/lightblue-mongo/pull/135#discussion_r35897457>
> :
>
> > @@ -79,8 +80,8 @@ public void setup() throws Exception {
> >
> >      @After
> >      public void teardown() throws Exception {
> > -        mongo.getConnection().dropDatabase(DATABADSE_NAME);
> > -        coll = null;
> > +        if(coll!=null)
> > +            coll.remove(new BasicDBObject());
>
> Not sure I understand this change. Why is this better?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lightblue-platform/lightblue-mongo/pull/135/files#r35897457>
> .
>